Source/Scene/PolygonClippingAccelerationGrid.js

```diff
@@ -0,0 +1,505 @@
+      rect.btmRight.x,
+      rect.btmRight.y,
+      ax,
+      ay,
+      cx,
+      cy
+    ) !== null
+  )
+    return true;
+  if (
+    lineIntersection(
+      rect.topLeft.x,
+      rect.topLeft.y,
+      rect.btmLeft.x,
+      rect.btmRight.y,
+      ax,
+      ay,
+      cx,
+      cy
+    ) !== null
+  )
+    return true;
+
+  return (
+    lineIntersection(
+      rect.topRight.x,
+      rect.topRight.y,
+      rect.btmRight.x,
+      rect.btmRight.y,
+      ax,
+      ay,
+      cx,
+      cy
+    ) !== null
+  );
+}
+
+function doesTriangleOverlapRect(rect, ax, ay, bx, by, cx, cy) {
+  var r = rect;
+
+  if (!pointInTriangle(r.topLeft.x, r.topLeft.y, ax, ay, bx, by, cx, cy)) {
+    return false;
+  }
+
+  if (!pointInTriangle(r.topRight.x, r.topRight.y, ax, ay, bx, by, cx, cy)) {
+    return false;
+  }
+
+  if (!pointInTriangle(r.btmRight.x, r.btmRight.y, ax, ay, bx, by, cx, cy)) {
+    return false;
+  }
+
+  return pointInTriangle(r.btmLeft.x, r.btmLeft.y, ax, ay, bx, by, cx, cy);
+}
+
+/**
+ * Construct a 2D bounding box from a provided triangulated
+ * mesh.
+ *
+ * @param {Object} options Object with the following properties:
+ * @param {Array.<Number>} options.positions An array of position data
+ * representing the clipping mesh. The bounding box is constructed around the
+ * XY axis, Z is ignored.
+ * @param {Array.<Number>} options.indices Triangle index data corresponding
+ * to the positions data.
+ *
+ * @private
+ */
+
+function ClippingBoundingBox2D(options) {
+  var positions = options.positions;
+  var indices = options.indices;
+
+  var minX = Infinity;
+  var minY = Infinity;
+  var maxX = -Infinity;
+  var maxY = -Infinity;
+
+  var i;
+  for (i = 0; i < indices.length; i += 3) {
+    var ax = positions[indices[i] * 3];
+    var ay = positions[indices[i] * 3 + 1];
+    var bx = positions[indices[i + 1] * 3];
+    var by = positions[indices[i + 1] * 3 + 1];
+    var cx = positions[indices[i + 2] * 3];
+    var cy = positions[indices[i + 2] * 3 + 1];
+
+    minX = Math.min(ax, minX);
+    minX = Math.min(bx, minX);
+    minX = Math.min(cx, minX);
+    minY = Math.min(ay, minY);
+    minY = Math.min(by, minY);
+    minY = Math.min(cy, minY);
+
+    maxX = Math.max(ax, maxX);
+    maxX = Math.max(bx, maxX);
+    maxX = Math.max(cx, maxX);
+    maxY = Math.max(ay, maxY);
+    maxY = Math.max(by, maxY);
+    maxY = Math.max(cy, maxY);
+  }
+
+  this.topLeft = { x: minX, y: maxY };
+  this.topRight = { x: maxX, y: maxY };
+  this.btmRight = { x: maxX, y: minY };
+  this.btmLeft = { x: minX, y: minY };
+  this.width = this.topRight.x - this.topLeft.x;
+  this.height = this.topRight.y - this.btmRight.y;
+}
+
+ClippingBoundingBox2D.prototype.toClockwiseCartesian2Pairs = function () {
+  return [
+    new Cartesian2(this.topLeft.x, this.topLeft.y),
+    new Cartesian2(this.topRight.x, this.topRight.y),
+    new Cartesian2(this.btmRight.x, this.btmRight.y),
+    new Cartesian2(this.btmLeft.x, this.btmLeft.y),
+  ];
+};
```

Comment:
Might be simpler to just compute this in the constructor and add a getter for it